feature/text_based.py

```diff
@@ -0,0 +1,703 @@
+
+       Parameters
+       ----------
+        selection_size: int
+            Maximum number of content to select.
+        seed: int, default=123456
+            Random seed.
+        trials: int, default=10
+            Number of trials to run for random functions to estimate average coverage.
+        verbose: bool, default=False
+            Print intermediate results if True.
+        """
+
+        # Validate arguments
+        self.featurization_method = None
+        self.cost_metric = None
+        self._validate_args(selection_size)
+
+        # Set input arguments
+        self.selection_size = selection_size
+        self.seed = seed
+        self.trials = trials
+        self.verbose = False
+
+        # Initialize class variables
+        self.matrix = None
+        self.features = None
+        self._num_rows = None
+        self._num_cols = None
+
+    def run_content_selection(self, input_df: pd.DataFrame, categories: pd.DataFrame, selection_size: int,
+                              featurization_method: TextWiser, optimization_method: str = "exact",
+                              cost_metric: str = "diverse", trials: int = 10) -> List:
+
+        """Run content selection algorithm.
+
+        Parameters
+        ----------
+        input_df: pd.DataFrame
+            Input data frame with categories and features of content to select from.
+
+        categories: pd.DataFrame
+            Data frame with data that contains categories/labels to be covered.
+
+        selection_size: int
+            Number of feature to select.
+
+        featurization_method: TextWiser
+
+        optimization_method: str, default="exact"
+            Optimization method used to perform content selection.
+                - See TextBased in Selector.py for supported options definition
+
+        cost_metric: str, default = "diverse"
+
+        trials: int, default = 10
+
+        Returns
+        -------
+        List of indices in data that are selected.
+        """
+
+        # Initialize arguments
+        self.selection_size = selection_size
+        self.featurization_method = featurization_method
+        self.cost_metric = cost_metric
+
+        # Process data
+        self._process_data(input_df, categories, selection_size, optimization_method, cost_metric)
+
+        if optimization_method == "exact":
+            selected = self._select_multi_level_optimization()
+        elif optimization_method == "greedy":
+            selected = self._select_greedy()
+        elif optimization_method == "kmeans":
+            selected = self._select_kmeans()
+        elif optimization_method == "random":
+            selected = self._select_random(trials=trials)
+        else:
+            raise NotImplementedError(f"{optimization_method} has not been implemented.")
+
+        # Sort selection
+        selected.sort()
+
+        return selected
+
+    def _process_data(self, input_df: pd.DataFrame, categories: pd.DataFrame, selection_size: int,
```

Comment:
I guess we should not call this input parameter as "categories" anymore?

Reply:
Fixed